### PR TITLE
hardening/1368_style_in_mongoBackend_04

### PR DESCRIPTION
**Pull request by kzangeli**
style guide in MongoCommonUpdate and compoundValueBson

### CODE REVIEW

File: src/lib/mongoBackend/compoundValueBson.h [hunk 1]
```diff
@@ -37,14 +37,14 @@
 *
 * compoundValueBson (for objects) -
 */
-extern void compoundValueBson(std::vector<orion::CompoundValueNode*> children, mongo::BSONObjBuilder& b);
+extern void compoundValueBson(const std::vector<orion::CompoundValueNode*>& children, mongo::BSONObjBuilder& b);
```
Comment (fgalan) [inline src/lib/mongoBackend/compoundValueBson.h]: Different from the general rule with style fixing, maybe this kind of change (X -> const X&) should be mentioned in the CNR, e.g.:

"Fix: several changes in argument passing in mongoBackend library to align style, from X to const X&"

File: src/lib/mongoBackend/MongoCommonUpdate.cpp [hunk 2]
```diff
@@ -130,15 +127,12 @@ static bool hasMetadata(std::string name, std::string type, ContextAttribute* ca
...
 
 
-
 /* ****************************************************************************
 *
 * equalMetadataValues -
-*
 */
-static bool equalMetadataValues(BSONObj& md1, BSONObj& md2)
+static bool equalMetadataValues(const BSONObj& md1, const BSONObj& md2)
```
Comment (fgalan) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Different from the general rule with style fixing, maybe this kind of change (X& -> const X&) should be mentioned in the CNR, e.g.:

"Hardening: several changes in argument passing in mongoBackend library to align style, from X& to const X&"

File: src/lib/mongoBackend/compoundValueBson.h (hunk 1 above)
Comment (kzangeli) [inline src/lib/mongoBackend/compoundValueBson.h]: It is true this is an actual change in code, not just style ...
However it is not a bug fix, *only* a fix to save space on the stack.
Do we really want to go this low level in CNR ?

Comment (fgalan) [inline src/lib/mongoBackend/compoundValueBson.h]: > However it is not a bug fix, only a fix to save space on the stack.

That's why I say to use "Hardening:" and not "Fix:" in the CNR line ;)

"Low level will" be going change by change mentioning every changed function in CNR. Adding a general comment like the one I proposed doesn't hurt.

Comment (kzangeli) [inline src/lib/mongoBackend/compoundValueBson.h]: Sure, no probs.
Just wanted to make sure you had thought it over ... :-)

Comment (kzangeli) [inline src/lib/mongoBackend/compoundValueBson.h]: Fixed in 3b07d1c

File: src/lib/mongoBackend/MongoCommonUpdate.cpp (hunk 2 above)
Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Fixed in 3b07d1c

Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Actually: c56d91e

File: src/lib/mongoBackend/compoundValueBson.h (hunk 1 above)
Comment (kzangeli) [inline src/lib/mongoBackend/compoundValueBson.h]: Actually: c56d91e

**Comment by fgalan** [comment]
LGTM

**Pull request merged**